Core/clim-core/presentations/typed-input.lisp

```diff
@@ -32,15 +32,11 @@
                          object type)))))
 
 (defun input-context-wait-test (stream)
-  (when-let ((event (event-peek stream)))
-    (let ((sheet (event-sheet event)))
-      (and (output-recording-stream-p sheet)
-           (typep event '(or pointer-event keyboard-event))))))
+  (when-let ((gesture (stream-gesture-available-p stream)))
+    (output-recording-stream-p (event-sheet gesture))))
```

Comment:
Can this result in `event-sheet` being called on a `character`?

Reply:
it can, good catch. fixed.